lib/azure/armrest/storage/managed_storage_helper.rb

```diff
@@ -116,8 +115,10 @@ def get_blob_raw(disk_name, resource_group = configuration.resource_group, optio
         retry
       end
     ensure
-      end_url = build_url(resource_group, disk_name, 'EndGetAccess')
-      rest_post(end_url)
+      if begin_get_access_response
```

Comment:
In what case `begin_get_access_response` is `nil`?

Reply:
Then we can't call EndGetAccess because BeginGetAccess didn't succeed.

Reply:
rest_post will return nil if an exception is raised. Though, I just realized that our ensure block is not on the method, but on the first begin block. Fixing...